doc/stdenv/stdenv.chapter.md

```diff
@@ -1136,6 +1136,29 @@ Example removing all references to the compiler in the output:
 }
 ```
 
+### `applyPatches` [ `-p` \<patch-set\> ... ] [--] [ \<additional-arguments\> ... ] {#fun-applyPatches}
+
+Applies a set of patches to the current directory. Patch files are automatically
+decompressed using `gzip` (`.gz`), `bzip2` (`.bz2`) or `xz` (`.xz`, `.lzma`)
+based on the file name extension. Additional arguments are passed directly to
+the `patch` command.
+
+A patch set is either a path to an individual patch file or a directory containing
+multiple patch files. If a directory contains a `series` file, it is treated as
+a list of patch files to apply. The `series` file lists patch file paths, one
+per line, relative to the patch set directory. Empty lines and comments (lines
+starting with `#`) are ignored. It is an error if a line starts with either `-`
+or `+`.
+
+When processing a directory without a `series` file, only files with standard
+patch extensions `.patch` and `.diff` will be considered for application,
+optionally with compressed file name extension (e.g. `.patch.gz`).
```

Comment:
See https://github.com/NixOS/nixpkgs/pull/335579#discussion_r1721978177

I agree with https://github.com/NixOS/nixpkgs/pull/335579#issuecomment-2296789754, I’ll update `applyPatches` to return an error if `series` file does not exist to avoid implicit ordering.

Reply:
Requiring a series file addresses this possibility definitively. I can get on board with that.